tests/autotune/test_autotune.py

```diff
@@ -0,0 +1,324 @@
+        )
+    }
+    model_config = {"dropout_rate": loguniform(1e-4, 1e-1)}
+    plan_config = {"lr": loguniform(1e-4, 1e-1)}
+    categorical_covariates = ["categorical_cov"]
+    num_epochs = 2
+    tuner = Autotune(
+        adata,
+        SCVI,
+        metric_functions=metric_functions,
+        training_metrics=metrics,
+        model_hyperparams=model_config,
+        plan_hyperparams=plan_config,
+        categorical_covariates=categorical_covariates,
+    )
+    asha_scheduler = ASHAScheduler(max_t=num_epochs, grace_period=1, reduction_factor=2)
+    model, analysis = tuner.run(
+        metric="silhouette_score",
+        mode="max",
+        scheduler=asha_scheduler,
+        local_dir=save_path,
+    )
+
+
+def test_hvg_filter(save_path):
+    adata = synthetic_iid()
+    metrics = [
+        "elbo_validation",
+        "reconstruction_loss_validation",
+    ]
+    metric_functions = {
+        "silhouette_score": partial(
+            silhouette_metric_labels_batch, labels_key="labels", batch_key="batch"
+        )
+    }
+    model_config = {"dropout_rate": loguniform(1e-4, 1e-1)}
+    plan_config = {"lr": loguniform(1e-4, 1e-1)}
+    num_epochs = 2
+    tuner = Autotune(
+        adata,
+        SCVI,
+        metric_functions=metric_functions,
+        training_metrics=metrics,
+        model_hyperparams=model_config,
+        plan_hyperparams=plan_config,
+        test_effect_hvg=True,
+        top_hvg=[1000, 2000],
+    )
+    asha_scheduler = ASHAScheduler(max_t=num_epochs, grace_period=1, reduction_factor=2)
+    model, analysis = tuner.run(
+        metric="silhouette_score",
+        mode="max",
+        scheduler=asha_scheduler,
+        local_dir=save_path,
+    )
+
+
+def test_hvg_no_filter(save_path):
+    adata = synthetic_iid()
+    metrics = [
+        "elbo_validation",
+        "reconstruction_loss_validation",
+    ]
+    metric_functions = {
+        "silhouette_score": partial(
+            silhouette_metric_labels_batch, labels_key="labels", batch_key="batch"
+        )
+    }
+    model_config = {"dropout_rate": loguniform(1e-4, 1e-1)}
+    plan_config = {"lr": loguniform(1e-4, 1e-1)}
+    num_epochs = 2
+    tuner = Autotune(
+        adata,
+        SCVI,
+        metric_functions=metric_functions,
+        training_metrics=metrics,
+        model_hyperparams=model_config,
+        plan_hyperparams=plan_config,
+        top_hvg=[1000, 2000],
+    )
+    asha_scheduler = ASHAScheduler(max_t=num_epochs, grace_period=1, reduction_factor=2)
+    model, analysis = tuner.run(
+        metric="silhouette_score",
+        mode="max",
+        scheduler=asha_scheduler,
+        local_dir=save_path,
+    )
+
+
+def test_complex(save_path):
+    adata = synthetic_iid()
+    adata.obs["continious_cov"] = np.linspace(0, 1, adata.n_obs)
+    adata.obs["categorical_cov"] = pd.Categorical(
+        ["cov_1", "cov_2"] * (round(adata.n_obs / 2))
+    )
+    metrics = [
```

Comment:
There's a lot of repeated code in this file. Could you move out some constants like metric_functions, model_config, plan_config to the top of the file since they are consistent across all tests?